cpp-inference/inference_scaling_imagenet.cpp

```diff
@@ -89,22 +91,18 @@ void run_mnist(const std::string& model_name,
       std::cout<<"Setting on device: "<< device <<std::endl<<std::flush;
       std::cout<<"Setting on "<< std::to_string(num_devices) << " devices" <<std::endl<<std::flush;
 
-      for (int i=0; i<num_devices; i++ ) {
-        std::string model_key = "resnet_model_" + std::to_string(i);
-        std::string script_key = "resnet_script_" + std::to_string(i);
-        std::string device_key = device + ":" + std::to_string(i);
-        std::cout<<"Device Key " <<device_key <<std::endl<<std::flush;
-
-        if (use_multi) {
-          client.set_model_from_file_multigpu(model_key, "./resnet50.pt", "TORCH", 0, num_devices, batch_size);
-          client.set_script_from_file_multigpu(script_key, "./data_processing_script.txt", 0, num_devices);
-	      }  
-        else {
-          client.set_model_from_file(model_key, "./resnet50.pt", "TORCH", device_key, batch_size);
-          client.set_script_from_file(script_key, device_key, "./data_processing_script.txt");
-        }
-        
+      std::string model_key = "resnet_model";
+      std::string script_key = "resnet_script";
+      std::string model_name = "./resnet50." + device + ".pt";
+      if (use_multi) {
+        client.set_model_from_file_multigpu(model_key, model_name, "TORCH", 0, num_devices, batch_size);
+        client.set_script_from_file_multigpu(script_key, "./data_processing_script.txt", 0, num_devices);
+      }  
+      else {
+        client.set_model_from_file(model_key, model_name, "TORCH", device, batch_size);
+        client.set_script_from_file(script_key, device, "./data_processing_script.txt");
```

Comment:
Is this hurting the CPU case by making all processors share the same model/script?

Reply:
OK, my bad, I did not think we wanted to support "multi-device" with CPU, I'll revert some of the changes. I think `use_multi` was not consistent between `driver.py` and the C++ code, then